anywhere-fitness-fe/src/components/ClientDashboard.js

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useEffect } from "react";
+import axiosWithAuth from "../utils/axiosWithAuth";
+import ClassCard from "./ClassCard";
+
+const ClientDashboard = () => {
+    const [allClasses, setAllClasses] = useState([]);
+    const [myClasses, setMyClasses] = useState([]);
+    const [refresh, setRefresh] = useState(false);
+    useEffect(() => {
+        axiosWithAuth()
+            .get("/client/classes")
+            .then((res) => {
+                console.log(res.data.data);
+                setMyClasses(res.data.data);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    }, [refresh]);
+
+    return (
+        <>
+            <button>Create a Class</button>
+            <h1>My Classes</h1>
+            <div class="classList">
+                {myClasses.map((item) => {
+                    return <ClassCard key={item.id} details={item}></ClassCard>;
+                })}
+            </div>
+        </>
+    );
+};
```

Comment:
I will use this section to demonstrate what I was talking about. The combination of useState and useEffect are everywhere. Following the single responsibility principle says we can abstract that logic to a reusable function.

A second concept I would like to share with you is the View Model. We use them everywhere in TypeScript. It makes it easier to design a single interface and make sure we always get everything we need for the page to render.

```
// note the export. If this api call is useful on any other page you can import it
// As long as this is the only possible use, keep it in this file.
// These could also be stored in the API folder so you have a central place for all your
// API calls so there is 1 place to refactor if the API changes, or you have to find another provider for the API
export const useFetchClasses = () => {
    const [allClasses, setAllClasses] = useState([]);
    const [myClasses, setMyClasses] = useState([]);
    const [refresh, setRefresh] = useState(false);
    useEffect(() => {
        axiosWithAuth()
            .get("/client/classes")
            .then((res) => {
                console.log(res.data.data);
                setMyClasses(res.data.data);
            })
            .catch((err) => {
                console.log(err);
                // Propper error logging is critical in a cloud application. It's usage
                // is typically boiled down to be as simple as console.log()
                // How would you implement a logger function like this to emit the error to some service?
                logAppInsightsError(err);
            });
    }, [refresh]);
    return {allClasses, myClasses, refresh} as ViewModel;
}
```

USAGE
```
const ClientDashboard = () => {
    const VM = useFetchClasses();
    return (<MyJSX />)
```
Proper levels of abstraction and meaningful variable names make your code read like a book